typescript/packages/subsurface-viewer/src/layers/grid3d/privateLayer.ts

```diff
@@ -258,7 +258,7 @@ export default class privateLayer extends Layer<privateLayerProps> {
 
         const vertexIndex = 256 * 256 * r + 256 * g + b;
 
-        if (info.coordinate?.[2]) {
+        if (typeof info.coordinate?.[2] !== "undefined") {
```

Comment:
This was correct: no need to invert 0.

The fix is to move line 265 (ie. layer_properties.push(...)) outside the if block (ie. after line 266)